src/ConnectedMode.UnitTests/Shared/SharedBindingConfigFileProviderTests.cs

```diff
@@ -0,0 +1,211 @@
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using System.IO;
+using System.IO.Abstractions;
+using SonarLint.VisualStudio.ConnectedMode.Shared;
+using SonarLint.VisualStudio.Core;
+using SonarLint.VisualStudio.TestInfrastructure;
+
+namespace SonarLint.VisualStudio.ConnectedMode.UnitTests.Shared
+{
+    [TestClass]
+    public class SharedBindingConfigFileProviderTests
+    {
+        [TestMethod]
+        public void MefCtor_CheckIsExported()
+        {
+            MefTestHelpers.CheckTypeCanBeImported<SharedBindingConfigFileProvider, ISharedBindingConfigFileProvider>(
+                MefTestHelpers.CreateExport<ILogger>());
+        }
+
+        [TestMethod]
+        public void ReadSharedBindingConfigFile_SQConfig_Reads()
+        {
+            var configFileContent = @"{""SonarQubeUri"":""https://127.0.0.1:9000"",""ProjectKey"":""projectKey""}";
+            var filePath = "Some Path";
+            var uri = new Uri("https://127.0.0.1:9000");
+
+            var logger = new TestLogger();
+            var file = CreateFile(filePath, configFileContent);
+            var fileSystem = GetFileSystem(file.Object);
+
+            var testSubject = CreateTestSubject(fileSystem.Object);
+
+            var result = testSubject.ReadSharedBindingConfigFile(filePath);
+
+            result.Uri.Should().Be("https://127.0.0.1:9000");
+            result.ServerUri.Should().BeEquivalentTo(uri);
+            result.ProjectKey.Should().Be("projectKey");
+            result.Organization.Should().BeNull();
+        }
+
+        [TestMethod]
+        public void ReadSharedBindingConfigFile_SCConfig_Reads()
+        {
+            string configFileContent = @"{""SonarCloudOrganization"":""Some Organisation"",""ProjectKey"":""projectKey""}";
+            string filePath = "Some Path";
+
+            var file = CreateFile(filePath, configFileContent);
+            var fileSystem = GetFileSystem(file.Object);
+
+            var testSubject = CreateTestSubject(fileSystem.Object);
+
+            var result = testSubject.ReadSharedBindingConfigFile(filePath);
+
+            result.Organization.Should().Be("Some Organisation");
+            result.ProjectKey.Should().Be("projectKey");
+            result.Uri.Should().BeNull();
```

Comment:
Setting the default Uri for SC would make code that uses this a bit less complicated. Not sure if it should be set by the file provider or by the config provider later.